Rules/CommonScripts/ChatCommands/PlayerCommands.as

```diff
@@ -0,0 +1,210 @@
+#include "ChatCommand.as"
+
+class ClassCommand : ChatCommand
+{
+	ClassCommand()
+	{
+		super("class", "Change your class");
+		SetUsage("<name>");
+	}
+
+	bool canPlayerExecute(CPlayer@ player)
+	{
+		return (
+			ChatCommand::canPlayerExecute(player) &&
+			!ChatCommands::getManager().whitelistedClasses.empty()
+		);
+	}
+
+	void Execute(string[] args, CPlayer@ player)
+	{
+		if (!isServer()) return;
+
+		CBlob@ blob = player.getBlob();
+		if (blob is null)
+		{
+			server_AddToChat(getTranslatedString("Your class cannot be changed while dead or spectating"), ConsoleColour::ERROR, player);
+			return;
+		}
+
+		if (args.size() == 0)
+		{
+			string[] classes = ChatCommands::getManager().whitelistedClasses;
+			server_AddToChat(getTranslatedString("Specify a class to swap to: " + join(classes, ", ")), ConsoleColour::ERROR, player);
+			return;
+		}
+
+		string className = args[0];
+
+		if (!isClassWhitelisted(className))
```

Comment:
Moderators should be able to bypass the class whitelist and choose whatever blob they want (assuming modding purposes here)
```suggestion
		if (!isClassWhitelisted(className) && !player.isMod())
```